packages/vike-node/src/runtime/vike-handler.ts

```diff
@@ -33,24 +33,89 @@ async function renderPage<PlatformRequest>({
   return pageContext.httpResponse
 }
 
-async function renderPageWeb<PlatformRequest>({
-  url,
-  headers,
-  platformRequest,
-  options
-}: {
-  url: string
-  headers: [string, string][]
-  platformRequest: PlatformRequest
-  options: VikeOptions<PlatformRequest>
-}) {
-  const httpResponse = await renderPage({
-    url,
-    headers,
-    platformRequest,
-    options
+export const renderPageCompress = ((options?) => async (request, _context, runtime: any) => {
+  const nodeReq: IncomingMessage | undefined = runtime.req
+  const compressionType = options?.compress ?? !isVercel()
+  const compressMiddleware = compressMiddlewareFactory()(request)
+
+  return async (response) => {
+    if (!globalStore.isPluginLoaded && nodeReq) {
+      const isAsset = nodeReq.url?.startsWith('/assets/')
+      const shouldCompressResponse = compressionType === true || (compressionType === 'static' && isAsset)
+      if (shouldCompressResponse) {
+        return compressMiddleware(response)
+      }
+    }
+    return response
+  }
+}) satisfies Get<[options: VikeOptions], UniversalMiddleware>
+
+export const renderPageHandler = ((options?) => async (request, context, runtime: any) => {
+  const nodeReq: IncomingMessage | undefined = runtime.req
+  let staticConfig: false | { root: string; cache: boolean } = false
+  let staticMiddleware: ConnectMiddleware | undefined
+
+  if (nodeReq) {
+    globalStore.setupHMRProxy(nodeReq)
+    const { resolveStaticConfig } = await import('./utils/resolve-static-config.js')
+    staticConfig = resolveStaticConfig(options?.static)
+  }
+
+  if (globalStore.isPluginLoaded) {
+    const handled = await web(request)
+
+    if (handled) return handled
+  } else if (nodeReq) {
+    if (staticConfig) {
+      const handled = await connectToWeb(serveStaticFiles)(request)
+      if (handled) return handled
+    }
+  }
+
+  async function serveStaticFiles(req: IncomingMessage, res: ServerResponse): Promise<boolean> {
+    if (!staticMiddleware) {
+      const { default: sirv } = await import('sirv')
+      staticMiddleware = sirv((staticConfig as { root: string; cache: boolean }).root, { etag: true })
```

Comment:
In itself I like the idea to have a built-in static server. Just not sure this is the right place. But let's keep it since it was already here before this PR.

In the long term (after this PR), what could be lovely is to leverage the upcoming partial eject functionality: upon running `$ eject vike-hono entry` the Hono entry is ejected, giving the user the opportunity to switch out the static server. In other words, the user can use a Hono server without writing a single line of server code. The idea here is that `vike-hono`, `vike-telefunc`, `vike-auth`, ... would all automatically integrate with each other. To swap the server all the user needs to do is to install `vike-express` instead of `vike-hono` (assuming he didn't write/eject any server code) and everything would work equally.

Reply:
I agree

Reply:
I guess we can resolve this.